pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -209,20 +190,18 @@ static bool proc_t_comm_equals(proc_t a, proc_t b) {
 }
 
 __attribute__((always_inline))
-static int are_offsets_ready_v4(tracer_status_t* status, struct sock* skp) {
+static bool check_family(struct sock* sk, tracer_status_t* status, u16 expected_family) {
+    u16 family = 0;
+    bpf_probe_read(&family, sizeof(u16), ((char*)sk) + status->offset_family);
+    return family == expected_family;
+}
+
+__attribute__((always_inline))
+static int guess_offsets(tracer_status_t* status, struct sock* skp) {
     u64 zero = 0;
 
-    switch (status->state) {
-    case TRACER_STATE_UNINITIALIZED:
-        return 0;
-    case TRACER_STATE_CHECKING:
-        break;
-    case TRACER_STATE_CHECKED:
-        return 0;
-    case TRACER_STATE_READY:
+    if (status->state != TRACER_STATE_CHECKING) {
```

Comment:
Is this the default value? What happens if another program calls `tcp_stats` before we've had a chance to setup our guessing

Reply:
Not enough coffee. `status` will be `NULL` in that case, so we should be fine, right?

Reply:
Humm, sorry, I'm not sure if I follow. Basically `guess_offsets` is now only called from 2 places (`kretprobe__tcp_v6_connect` and `kprobe__tcp_get_info`) and both are used exclusively for offset guessing. All other kprobes have their handlers attached only after the offset guessing process is done, so that's why we don't even have to bother about the `status.state` anymore in the rest of the code.

Reply:
Oh I think I misunderstood your message. Yes, in that case `status` will be `NULL`.